tla-bmcmt/src/main/scala/at/forsyte/apalache/tla/bmcmt/config/CheckerExceptionAdapter.scala

```diff
@@ -61,10 +61,12 @@ class CheckerExceptionAdapter @Inject() (sourceStore: SourceStore, changeListene
       NormalErrorMessage("%s: Input error (see the manual): %s".format(findLoc(err.causeExpr.ID), err.getMessage))
 
     case err: TypingInputException =>
-      NormalErrorMessage("Typing input error: " + err.getMessage)
+      // this is a normal message, as we know that the typing error is due to the user
+      NormalErrorMessage("%s: type input error: %s".format(findLoc(err.causeExprId), err.getMessage))
 
     case err: TypingException =>
-      FailureMessage("Type checker error: " + err.getMessage)
+      // this is a failure message, as we know that something type-related in apalache is broken
+      FailureMessage("%s: type error: %s".format(findLoc(err.causeExprId), err.getMessage))
```

Comment:
Should the message reflect the comment? Perhaps something like:

```suggestion
      FailureMessage("%s: internal error in type checking: %s".format(findLoc(err.causeExprId), err.getMessage))
```